graph/pull_v2.go

```diff
@@ -223,6 +223,9 @@ func (p *v2Puller) pullV2Tag(tag, taggedName string) (verified bool, err error)
 	go func() {
 		if _, err := io.Copy(out, pipeReader); err != nil {
 			logrus.Errorf("error copying from layer download progress reader: %s", err)
+			if err := pipeReader.CloseWithError(err); err != nil {
```

Comment:
No, that defer is only reached after pull has completed(or real download error). The goal is to carry on with the pull although client has disconnected.